python/lsst/ctrl/oods/butlerAttendant.py

```diff
@@ -110,11 +114,98 @@ async def ingest(self, file_list):
         if self.s3profile:
             # rewrite URI to add s3profile
             new_list = [s.replace(netloc=f"{self.s3profile}@{s.netloc}") for s in file_list]
+
+        raws = []
+        for entry in new_list:
+            if type(entry) is str:
```

Comment:
ResourcePath is immutable so returns itself if you try to construct a new one from one, so this code should be the same:
```python
for entry in new_list:
    rp = ResourcePath(entry)
    if rp.path.endswith("_guider.fits"):
        self.guiders.append(entry)
    else:
        raws.append(entry)
```
but may be easier to make your test code use ResourcePath?

Reply:
Probably good to change the test code.  I'll add that to another ticket.  In the meantime I made the change you suggested above.